client/landing/gutenboarding/gutenboard.tsx

```diff
@@ -44,13 +53,23 @@ const toggleSidebarShortcut = {
 
 registerBlockType( name, settings );
 
-const onboardingBlock = createBlock( name, {} );
-
 export function Gutenboard() {
 	const [ isEditorSidebarOpened, updateIsEditorSidebarOpened ] = useState( false );
-
 	const toggleGeneralSidebar = () => updateIsEditorSidebarOpened( isOpen => ! isOpen );
 
+	const onboardingState = useSelect( select => select( STORE_KEY ).getState() );
+
+	// FIXME: Quick'n'dirty step state, replace with router
+	const [ currentStep, setStep ] = useState( Steps.IntentGathering );
+	const goToNextStep = stepCompleted[ currentStep ]( onboardingState )
+		? () => setStep( step => step + 1 )
+		: undefined;
+	const goToPrevStep = currentStep > 0 ? () => setStep( step => step - 1 ) : undefined;
+
+	const onboardingBlock = useMemo( () => createBlock( name, { step: currentStep } ), [
+		currentStep,
+	] );
```

Comment:
`useMemo` doesn't have any lifetime guarantees, it's an optimization. To get a stable reference that changes with `currentStep`, we'll likely need `useRef` + `useEffect`.

#38396